eta/core/video.py

```diff
@@ -108,6 +116,686 @@ def glob_videos(dir_):
+        '''Adds the list of GPSWaypoints to this object.'''
+        if not self.has_gps:
+            self.gps_waypoints = []
+        self.gps_waypoints.extend(waypoints)
+        self._init_gps()
+
+    def get_timestamp(self, frame_number=None, world_time=None):
+        '''Gets the timestamp for the given point in the video.
+
+        Exactly one keyword argument must be supplied.
+
+        Args:
+            frame_number: the frame number of interest
+            world_time: a datetime describing the world time of interest
+
+        Returns:
+            the timestamp (in seconds) in the video
+        '''
+        if world_time is not None:
+            return world_time_to_timestamp(world_time, self.start_time)
+
+        return frame_number_to_timestamp(
+            frame_number, self.total_frame_count, self.duration)
+
+    def get_frame_number(self, timestamp=None, world_time=None):
+        '''Gets the frame number for the given point in the video.
+
+        Exactly one keyword argument must be supplied.
+
+        Args:
+            timestamp: the timestamp (in seconds) of interest
+            world_time: a datetime describing the world time of interest
+
+        Returns:
+            the frame number in the video
+        '''
+        if world_time is not None:
+            return world_time_to_frame_number(
+                world_time, self.start_time, self.duration,
+                self.total_frame_count)
+
+        return timestamp_to_frame_number(
+            timestamp, self.duration, self.total_frame_count)
+
+    def get_gps_location(
+            self, frame_number=None, timestamp=None, world_time=None):
+        '''Gets the GPS location at the given point in the video.
+
+        Exactly one keyword argument must be supplied.
+
+        Nearest neighbors is used to interpolate between waypoints, if
+        necessary.
+
+        Args:
+            frame_number: the frame number of interest
+            timestamp: the number of seconds into the video
+            world_time: a datetime describing the absolute (world) time of
+                interest
+
+        Returns:
+            the (lat, lon) at the given frame in the video, or None if the
+                video has no GPS waypoints
+        '''
+        if not self.has_gps:
+            return None
+        if world_time is not None:
+            timestamp = self.get_timestamp(world_time=world_time)
+        if timestamp is not None:
+            frame_number = self.get_frame_number(timestamp=timestamp)
+        return self._flat(frame_number), self._flon(frame_number)
+
+    def attributes(self):
+        _attrs = [
+            "start_time", "frame_size", "frame_rate", "total_frame_count",
+            "duration", "size_bytes", "encoding_str", "gps_waypoints"
+        ]
+        # Exclue attributres that are None
+        return [a for a in _attrs if getattr(self, a) is not None]
+
+    @classmethod
+    def build_for(
```

Comment:
Yeah probably but alas I've already written a handful of other classes that use `build_for`...

Reply:
We'll live with it for now, but noting it may come back to byte us.  Perhaps we have two different `build_for_X` and now they are both called `build_for` :)